src/main/java/com/example/Ecommerce/service/CartItemQueueService.java

```diff
@@ -0,0 +1,80 @@
+package com.example.Ecommerce.service;
+
+import com.example.Ecommerce.entity.*;
+import com.example.Ecommerce.exception.ResourceNotFoundException;
+import com.example.Ecommerce.model.QueueItem;
+import com.example.Ecommerce.repository.CartItemRepo;
+import com.example.Ecommerce.repository.ProductRepo;
+import com.example.Ecommerce.util.NotificationClient;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.PriorityQueue;
+
+@Service
+@RequiredArgsConstructor
+public class CartItemQueueService {
```

Comment:
I would name it `CartItemReservationService` or `CartItemReservationExpirationService`